lib/driver.rb

```diff
@@ -0,0 +1,76 @@
+require_relative 'csv_record'
+
+STATUSES = [:AVAILABLE, :UNAVAILABLE]
+
+module RideShare
+  class Driver < CsvRecord
+
+    attr_reader :name, :vin, :status, :trips
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE, trips: [])
+      super(id)
+      raise ArgumentError, 'Name must be a string' unless name.class == String
+      @name = name
+      raise ArgumentError, 'Incorrect VIN format' unless vin.length == 17 && vin.class == String
```

Comment:
We generally don't bother checking types/classes of arguments in Ruby.  The reason is that in Ruby we prefer the flexibility that dynamic types give us.  This would allow us later to create a `Vin` class and as long as it had a `length` and a useful `to_s` method things would continue to work.

If you're interested in version of Ruby that checks types before you run your code you can look at [Sorbet](https://sorbet.org/) (just a heads up though it is very verbose).

Also, you should have checked the type before calling `length` if you're going to check them.  (At the point you call `vin.length` you don't know that `vin` has a `length` method.)

```suggestion
      raise ArgumentError, 'Incorrect VIN format' unless vin.class == String && vin.length == 17
```